samples/drivers/spi_flash_at45/src/main.c

```diff
@@ -8,7 +8,7 @@
 #include <drivers/flash.h>
 #include <logging/log_ctrl.h>
 
-#define FLASH_DEVICE  "DATAFLASH_1"
+#define FLASH_DEVICE        DT_LABEL(DT_ALIAS(ext_flash0))
```

Comment:
This isn't really the right way to do it: it's not "wrong" but it's unnecessarily complex and could confuse users.  E.g. why are you declaring two devices in the overlay when the test only uses one?

Because the test here only uses one device, so there's no need to declare multiple ones, nor an alias.  Just use `DT_INST(0, atmel_at45)`.  The filter ensures there will be one.

When you do need to distinguish multiple instances the right way is to use node labels directly; there's no need for aliases.  See how it's done at https://github.com/zephyrproject-rtos/zephyr/blob/c54a511d26d0127f60ef6f2fbee6b3757defe627/tests/drivers/i2c/i2c_slave_api/src/main.c#L25.

Reply:
> E.g. why are you declaring two devices in the overlay when the test only uses one?

This is on the current sample, and, I don't know the why. It is one thing that obligates me to try DT_ALIAS.
I can do the change for only one instance, no problem.

@anangl , @nvlsianpu Is there any objection to change original overlay to use only one instance?

Reply:
> > E.g. why are you declaring two devices in the overlay when the test only uses one?
> 
> This is on the current sample, and, I don't know the why. It is one thing that obligates me to try DT_ALIAS.
> I can do the change for only one instance, no problem.
> 
> @anangl , @nvlsianpu Is there any objection to change original overlay to use only one instance?

Hi @anangl , @nvlsianpu did you have time to check if is possible change original overlay to use only one instance?